app/services/confirm_enrolment.rb

```diff
@@ -6,7 +6,7 @@ def initialize(application_choice:)
   end
 
   def call
-    @application_choice.update_attribute(:status, :enrolled)
+    @application_choice.state.confirm_enrolment!
```

Comment:
not sure about the use of `.state`,
I'd expect `@application_choice.state` to be the state of the application,  but actually `state` is the state machine itself.

Reply:
Any suggestions for an alternative?

Reply:
Does `ApplicationState` have to live inside `ApplicationChoice`? How about an interface like this (supposing class renamed to `ApplicationChoiceStatusChange`, above)

```
status_change = ApplicationChoiceStatusChange.new(application_choice)
status_change.confirm_enrolment!
```

or even this? (bit of programming required)

```
status_change = ApplicationChoiceStatusChange.new(application_choice, :confirm_enrolment!)
status_change.commit!
```